llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5078,10 +5151,22 @@ Function *SPIRVToLLVM::transFunction(SPIRVFunction *bf) {
   for (size_t i = 0, e = bf->getNumBasicBlock(); i != e; ++i) {
     SPIRVBasicBlock *bbb = bf->getBasicBlock(i);
     BasicBlock *bb = dyn_cast<BasicBlock>(transValue(bbb, f, nullptr));
+    BasicBlock *lastBb = bb;
+
     for (size_t bi = 0, be = bbb->getNumInst(); bi != be; ++bi) {
       SPIRVInstruction *bInst = bbb->getInst(bi);
+      if (continueAtLastInsertPoint) {
+        if (auto possibleNewBB = getBuilder()->GetInsertBlock()) {
+          BasicBlock *newBB = dyn_cast<BasicBlock>(possibleNewBB);
+          if (newBB && newBB != bb) {
+            bb = newBB;
+          }
+        }
+      }
       transValue(bInst, f, bb, false);
     }
+    continueAtLastInsertPoint = false;
```

Comment:
transValue() can call the function "transFunction()" again, so it can happen that continueAtLastInsertPoint is not cleared before you start translating another function.